examples/validator/index.ts

```diff
@@ -0,0 +1,56 @@
+import { LLMValidator } from "@/validator"
+import OpenAI from "openai"
+import { z } from "zod"
+
+const QuestionAnswerSchema = z.object({
+  question: z.string(),
+  answer: z.string()
```

Comment:
?is there a way to attach the validator to a property of answer

Reply:
hmm u mean like this right? 

```
class QuestionAnswerNoEvil(BaseModel):
    question: str
    answer: Annotated[
        str,
        BeforeValidator(
            llm_validator("don't say objectionable things", allow_override=True)
        ),
    ]
  ```
  
i am p sure yeah -- we can do like this in zod: 
```
const questionAnswerSchema = z.object({
  question: z.string(),
  answer: z.string().refine((answer) => {
 ... 
  }, {
    message: "custom error" 
  }),
});
```

But gotta go rn and can work on that tmrw.

Reply:
jk lol just added

Reply:
You can do it like so:
```
const QuestionAnswerSchemaNoEvil = z.object({
  question: z.string(),
  answer: z.string().refine(async answer => {
    const validator = new LLMValidator({
      statement: "don't say objectionable things",
      allowOverride: false,
      model: "gpt-3.5-turbo",
      temperature: 0,
      openaiClient: new OpenAI({
        apiKey: process.env.OPENAI_API_KEY ?? undefined,
        organization: process.env.OPENAI_ORG_ID ?? undefined
      })
    })
    await validator.validate(answer)
    return true
  })
})
```